docs/proposals/operator-sdk-v1-migration.md

```diff
@@ -0,0 +1,285 @@
+baked in CustomResource metrics, and another to serve our application-specific metrics.
+
+In v1 there is only one metrics port. The application exposes metrics on port 8080 by default.
+The generator then adds a kube-rbac-proxy sidecar to the controller manager, which acts as a
+reverse HTTP proxy for the metrics. With the rbac proxy, metrics are served over TLS and authorized
+via subject access reviews [13]. Baked in CustomResource metrics can be turned on via code.
+
+The generated services can all be tuned and adjusted via generated YAML manifests. The manifests
+are managed via kustomize to render the final deployments. Items like the rbac-proxy can be removed
+by adjusting the kustomize settings.
+
+It is unclear if OLM would tear down the old metrics service on our behalf.
+
+[12] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#export-metrics
+[13] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#use-handler-from-operator-lib
+
+### Process migrations
+
+A few of the common development commands have been moved out of operator-sdk:
+
+1. `operator-sdk generate crds` has been replaced with the generated `make manifests` target.
+2. `operator-sdk build` has likewise been replaced with a make target - `make docker-build` [13].
+3. Deploying is likewise handled by a make targets - `make install` and `make deploy` [14].
+
+The generated Makefile targets can be used by contributors to test Shipwright Builds. We are free
+to add our own targets to generate an "all-in-one" YAML manifest.
+
+[13] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#generate-manifests-and-build-the-operator
+[14] https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/
+
+
+### Risks and Mitigations
+
+**Risk**: Bug fixes and features are wiped out as we migrate to operator-sdk v1.
+
+*Mitigation*: As the rebase PR nears completion, we hold outstanding pull requests until it merges.
+Our CI needs to ensure that the rebase can merge without conflicts.
+
+## Design Details
+
+### Test Plan
+
+Existing test suites would remain in place, with overhauls to utilize envtest for integration testing.
+The generated `make test` target runs `go test ./...`, 
+
+### Graduation Criteria
+
+Not applicable
+
+### Upgrade / Downgrade Strategy
+
+Since this project is in the alpha state, support for in-place upgrade is not a hard requirement.
+This proposal outlines specific areas where there is potential to introduce breaking changes that
+downstream distributions may need to consider.
+
+### Version Skew Strategy
+
+If the leader election ConfigMap is left unchanged, we can support operator skew.
+
+## Implementation History
+
+2020-12-08: Initial proposal
+2020-12-09: Alternative using knative/pkg
+
+## Drawbacks
+
+Migrating to operator-sdk v1 requires significant project restructuring and changes to how we run
```

Comment:
Not sure if I agree here. I think @sbose78 setup the project with the sdk originally, but on a daily basis, the sdk is rarely used. We have very few places where we use it and those can be substitute with other artifacts. So I´m wondering if we need to contemplate what does the sdk provides is currently giving us?

Because doing the migration to v1 feels like a lot of changes, very prone to human error, and is not very clear what we are getting back from this?

Reply:
I added more info in the "Motivation" section regarding OLM. Bundle images are the preferred means of deploying operators managed by OLM.